### PR TITLE
Change base url

### LINKED ISSUE #62: invalid_server_details
After my HA was updated to latest 2024.5.4 version, integration stopped working..
I got error "invalid_server_details".
Tried re-configure it, remove and add, no luck.
With same email and password I can login to https://rym-pro.com site without problem.
In HACS my integration version was v2.0.14 , I've downloaded master 890e282, still no luck.
When I tried to add integration(after downloading master version in hacs and restarting HA) I got several  different errors:
 "Source: custom_components/citymind_water_meter/config_flow.py:69 . Failed to create integration, Error: 
invalid_server_details"
another error:
"Source: custom_components/citymind_water_meter/component/api/api.py:374
integration: CityMind Water Meter
Failed to access API
Invalid credentials, Error #5060: Email and/or password are not valid

another error.
Failed to post JSON to https://api.city-mind.com/consumer/login, HTTP Status: Too Many Requests (429), Data: {'error': 'Unauthorized request, user request rate exceeded limit.', 'code': 429}
But again with same credentials I can log in to website without errors(actually I still logged in and see my water consumption on the web site)

### PR DESCRIPTION
**Pull request by Arielgordon123**
closes #62

### CODE REVIEW

**Comment by elad-bar** [comment]
@Arielgordon123 thanks for the PR, have you checked it to work?
Reason I'm asking because the endpoints both old and new changed to the latest?

**Comment by elad-bar** [comment]
In addition, can you pls change the version number in manifest and update the changelog as part of this PR?

Thanks again

**Comment by Arielgordon123** [comment]
> @Arielgordon123 thanks for the PR, have you checked it to work? Reason I'm asking because the endpoints both old and new changed to the latest?

as far as i can see this is how it's working from the web-ui
except for the ENDPOINT_MY_MESSAGES that I've didnt see, and not working with the new url

**Comment by elad-bar** [comment]
@Arielgordon123 thank you for the efforts to make it work again, had few minutes to review the changes,
it seems that you changed the base url without fully testing that all endpoints are working,
to understand the range of supported endpoints, i strongly suggest to create a postman project and investigate the set of new endpoints within RYM pro website.

it's not that straight forward as it seems, some of the endpoints are not aligned with the changes, including monthly consumption and some of the endpoints of consumer / municipal,

if it's too much effort, i will try to get to it during the weekend and align it

thanks

File: custom_components/citymind_water_meter/component/helpers/const.py [hunk 1]
```diff
@@ -35,32 +34,31 @@
...
-ENDPOINT_CONSUMPTION_DAILY = f"{ENDPOINT_CONSUMPTION_NEW}/daily/lastbillingCycle/{{meter_id}}/{{yesterday}}/{{today}}"
-ENDPOINT_CONSUMPTION_MONTHLY = f"{ENDPOINT_CONSUMPTION_NEW}/monthly/{{meter_id}}/{{current_month}}/{{last_day_month}}"
-ENDPOINT_VACATIONS = f"{ENDPOINT_CONSUMER_OLD}/vacations"
-ENDPOINT_MY_ALERTS = f"{ENDPOINT_CONSUMER_NEW}/myalerts"
-ENDPOINT_MY_ALERTS_SETTINGS = f"{ENDPOINT_CONSUMER_OLD}/myalerts/settings"
+ENDPOINT_ALERTS_FOR_SETTINGS = f"{ENDPOINT_CONSUMER}/alertsForSettings"
+
+ENDPOINT_LAST_READ = f"{ENDPOINT_CONSUMPTION}/last-read"
+ENDPOINT_CONSUMPTION_DAILY = f"{ENDPOINT_CONSUMPTION}/daily/{{meter_id}}/{{yesterday}}/{{today}}"
+ENDPOINT_CONSUMPTION_MONTHLY = f"{ENDPOINT_CONSUMPTION}/monthly/{{meter_id}}/{{current_month}}/{{last_day_month}}"
```
Comment (elad-bar) [inline custom_components/citymind_water_meter/component/helpers/const.py:54]: Endpoint is not related to the default consumption but v1.1 of that endpoint, should be:
```python
ENDPOINT_CONSUMPTION_MONTHLY = f"{API_URL}/v1.1/consumption/monthly/{{meter_id}}/{{current_month}}/{{last_day_month}}"
```

**Comment by elad-bar** [comment]
> Is it worth merging this and then fixing the remaining broken URLs when possible? For those who don't take the update manually, the entire integration is now broken; this changes restores the vast majority of the most important functionality, which is a lot better than the current situation!

i prefer to release a version that is working and not one that partially working, monthly sensor wouldn't work properly if it was just released as suggested...
for you it will be fine, for others it will not be fine, i prefer to be bit more strict, hope it make sense

File: custom_components/citymind_water_meter/component/helpers/const.py (hunk 1 above)
Comment (Arielgordon123) [reply custom_components/citymind_water_meter/component/helpers/const.py:54]: the v1.1 api response is slightly different and will result error

without v1.1 the response is:
image

and with the v1.1 the response is:
image

and there is error 
`2024-05-23 16:05:26.700 ERROR (MainThread) [custom_components.citymind_water_meter.component.managers.home_assistant] Failed to load extract consumption (consumption-monthly) details, Meter {'meterCount': 11547, 'meterSn': '', 'fullAddress': '}, Error: 'str' object has no attribute 'get', Line: 904`

Comment (elad-bar) [reply custom_components/citymind_water_meter/component/helpers/const.py:54]: I checked in developer tools the request / response of the endpoint, it uses v1.1, i guess that's the right endpoint we need to use,
Based on past modifications they have conducted soon the non v1.1 endpoint will be removed

Comment (Arielgordon123) [reply custom_components/citymind_water_meter/component/helpers/const.py]: @elad-bar 
I've added now a new commit with the v1.1 endpoint for the monthly consumption

**Comment by elad-bar** [comment]
Thanks a lot!
Please let me know when I can merge it

**Comment by Arielgordon123** [comment]
I think I'm done with this Change if you don't have any suggestions

**Comment by elad-bar** [comment]
Looks good, assume you have tested it

**Pull request merged**